standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaHook.java

```diff
@@ -48,6 +48,9 @@
   public List<String> allowedAlterTypes = ImmutableList.of("ADDPROPS", "DROPPROPS");
   String ALTERLOCATION = "ALTERLOCATION";
   String ALLOW_PARTITION_KEY_CHANGE = "allow_partition_key_change";
+  String SET_PROPERTIES = "set_properties";
+  String UNSET_PROPERTIES = "unset_properties";
+  String PROPERTIES_SEPARATOR = "\n";
```

Comment:
Not sure which characters are unsupported in a key, but I think ' might work better as a separator due to the rules of the query language (since ' is used to mark the end of the table props in the SQL, e.g. 'key'='value')

Reply:
I like that. Adding `'` to the property key, even if it's escaped, will result in an exception during query parsing.